include/root_finders/quadratic_polynomial_solvers.hpp

```diff
@@ -0,0 +1,77 @@
+// VolEsti (volume computation and sampling library)
+// Copyright (c) 2021 Vissarion Fisikopoulos
+// Copyright (c) 2021 Apostolos Chalkis
+
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+
+#ifndef QUADRATIC_POLYNOMIAL_SOLVERS_H
+#define QUADRATIC_POLYNOMIAL_SOLVERS_H
+
+// The function returns the sigh of the input number
+template <typename T> int sgn(T val) 
+{
+    return (T(0) < val) - (val < T(0));
+}
+
+// The function compute the roots of a quadratic polynomial equation
+template <typename NT>
+void solve_quadratic_polynomial(NT a, NT b, NT c, NT &x1, NT &x2, bool &real) {
+
+    NT Delta = b * b - 4.0 * a * c;
+    if (Delta < NT(0)) {
+        real = false;
+        return;
+    }
+    real = true;
+    if (b >= NT(0)){
+        x1 = (- b - std::sqrt(Delta)) / (2.0 * a);
+        x2 = (2.0 * c) / (- b - std::sqrt(Delta));
+    } else {
+        x1 = (2.0 * c) / (- b + std::sqrt(Delta));
+        x2 = (- b + std::sqrt(Delta)) / (2.0 * a);
+    }
+}
+
+// The function compute the roots of a quadratic polynomial equation
+// using the algorithm in Revisiting the stability of computing the roots of a quadratic polynomial
+// by Nicola Mastronardi and Paul Michel Van Dooren, Electronic transactions on numerical analysis, 2014
+template <typename NT>
+void solve_qudratic_polynomial_stable(NT a, NT b, NT c, NT &x1, NT &x2, bool &real)
```

Comment:
this is currently not used anywhere in the code is this for testing reasons? It would be useful to add some test cases.

Reply:
Yes it is for test reasons. My experiments show that the method I use is more accurate than this one. I propose to remove the code since we dont use it.